package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -288,20 +288,27 @@ def _write_atoms(self, atoms):
         indices = atoms.indices + 1
         types = atoms.types.astype(np.int32)
 
+        try:
+            moltags = data['molecule_tag']
+        except KeyError:
+            moltags = [0] * len(atoms)
+
+
         if self.convert_units:
             coordinates = self.convert_pos_to_native(atoms.positions, inplace=False)
 
         if has_charges:
-            for index, atype, charge, coords in zip(indices, types, charges,
-                    coordinates):
-                self.f.write('{i:d} 0 {t:d} {c:f} {x:f} {y:f} {z:f}\n'.format(
-                             i=index, t=atype, c=charge, x=coords[0],
+            for index, moltag, atype, charge, coords in zip(indices, moltags,
+                    types, charges, coordinates):
+                self.f.write('{i:d} {m:d} {t:d} {c:f} {x:f} {y:f} {z:f}\n'.format(
+                             i=index, m=moltag, t=atype, c=charge, x=coords[0],
```

Comment:
I'm wondering if with this many fields, switching to f-strings would be cleaner?

```python
'{i:d} {m:d} {t:d} {c:f} {x:f} {y:f} {z:f}\n'.format(i=index, m=moltag, t=atype, c=charge, x=coords[0], y=coords[1], z=coords[2])
```
vs
```python
x, y, z = coords
"{index:d} {moltag:d} {atype:d} {charge:f} {x:f} {y:f} {z:f}\n"